src/app/services/http/gb-backend-http.service.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2019  The Hyve B.V.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {Injectable} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {HttpHelper} from 'app/utilities/http-helper';
+import {AppConfig} from '../../config/app.config';
+import {GbBackendQuery} from '../../models/gb-backend-models/gb-backend-query';
+import {Observable} from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GbBackendHttpService {
+
+  private httpHelper: HttpHelper;
+  private _endpointUrl: string;
+
+  constructor(private appConfig: AppConfig, httpClient: HttpClient) {
+    this.endpointUrl = `${this.appConfig.getConfig('gb-backend-url')}`;
+    this.httpHelper = new HttpHelper(this.endpointUrl, httpClient);
+  }
+
+  get endpointUrl(): string {
+    return this._endpointUrl;
+  }
+
+  set endpointUrl(value: string) {
+    this._endpointUrl = value;
+  }
+
+
+  // -------------------------------------- query calls --------------------------------------
+  /**
+   * Get the queries that the current user has saved.
+   * @returns {Observable<Query[]>}
+   */
+  getQueries(): Observable<GbBackendQuery[]> {
+    const urlPart = `queries`;
+    const responseField = 'queries';
+    return this.httpHelper.getCall(urlPart, responseField);
+  }
+
+  /**
+   * save a new query
+   * @param {GbBackendQuery} query
+   * @returns {Observable<GbBackendQuery>}
+   */
+  saveQuery(query: GbBackendQuery): Observable<GbBackendQuery> {
+    const urlPart = `queries`;
+    const queryBody = {};
```

Comment:
Why not just send the query object directly?

Reply:
I don't understand why if would be different, actually.

Reply:
you're right, this mapping seems not to be needed, I thought createDate/updateDate may cause some issues, but it works well. It's updated now.